apps/ledger-live-desktop/src/newArch/WalletSync/hooks/useTrustchainSdk.ts

```diff
@@ -13,10 +13,21 @@ export function runWithDevice<T>(
   return lastValueFrom(withDevice(deviceId || "")(transport => from(fn(transport))));
 }
 
-const defaultContext = { applicationId: 16, name: `LLD Sync ${randomUUID().slice(0, 8)}` }; // TODO : get name dynamically depending on the platform
+const platformMap: Record<string, string | undefined> = {
```

Comment:
Why undefined?

Reply:
`Record<string, F>` is rarely safe, we always want `Record<string, F | undefined>`

here is an example

![Capture d’écran 2024-07-08 à 16 33 13](https://github.com/LedgerHQ/ledger-live/assets/211411/25942001-38af-451f-97ce-c12f9055aa7c)


in our case, os.platform() can theorically be something else than linux,mac,windows.